app/src/main/java/ch/epfl/sdp/appart/PanoramaActivity.java

```diff
@@ -104,4 +107,94 @@ public void goBack(View view) {
         finish();
     }
 
+    /**
+     * Load previous panorama image
+     *
+     * @param view
+     */
+    public void goLeft(View view) {
+        currImage--;
+        loadImage();
+        if (currImage == 0)
+            disableLeftButton();
+        if (currImage == images.size() - 2)
+            enableRightButton();
+    }
+
+    /**
+     * Load next panorama image
+     *
+     * @param view
+     */
+    public void goRight(View view) {
+        currImage++;
+        loadImage();
+        if (currImage == images.size() - 1)
+            disableRightButton();
+        if (currImage == 1)
+            enableLeftButton();
+    }
+
+    /**
+     * Init the list of panorama images, set the current image index to 0, load image.
+     */
+    private void getImages() {
+        images = new ArrayList<>();
+        // TODO change with database call to get image references
+        images.add("file:///android_asset/panorama_test.jpg");
+        images.add("file:///android_asset/panorama_test_2.jpg");
+        images.add("file:///android_asset/panorama_test_3.jpg");
+        images.add("file:///android_asset/panorama_test_4.jpg");
+    }
+
+    /**
+     * Load image at current index into panorama
+     */
+    private void loadImage() {
+        PLSphericalPanorama panorama = new PLSphericalPanorama();
+
+        CompletableFuture<Bitmap> bitmapFuture = new CompletableFuture<>();
```

Comment:
I think this future needs an exceptionally handling lambda. So that if something fails, we can recover or display something to the user. Now that I think of it, we could have a class or something that handles UI for everything related to failure.

Reply:
good idea, I will add the failure handling, which was already missing from the code before this PR.